src/biogeophys/SoilFluxesMod.F90

```diff
@@ -205,6 +205,74 @@ subroutine SoilFluxes (bounds, num_urbanl, filter_urbanl, &
+         ! affected here are all related to the snow subgrid patch only because of the use of qflx_ev_snow.
+         ! In the situations where there are snow layers or there is snow without an explicit snow layer,
+         ! the partitioned variables will represent the components of snow evaporation
+         ! (qflx_ev_snow = qflx_liqevap_from_top_layer + qflx_solidevap_from_top_layer
+         ! - qflx_liqdew_to_top_layer - qflx_soliddew_to_top_layer).
+         ! In the case of no snow, qflx_ev_snow has already been set equal to qflx_ev_soil (the evaporation
+         ! from the subgrid soil patch) and the partitioned variables will then represent evaporation from the
+         ! subgrid soil patch.
+         ! In the case of urban columns (and lake columns - see LakeHydrologyMod), there are no subgrid
+         ! patches and qflx_evap_soi is used. qflx_evap_soi = qflx_liqevap_from_top_layer + qflx_solidevap_from_top_layer
+         ! - qflx_liqdew_to_top_layer - qflx_soliddew_to_top_layer.
+         if (.not. lun%urbpoi(l)) then
+            if (qflx_ev_snow(p) >= 0._r8) then
+               ! for evaporation partitioning between liquid evap and ice sublimation,
+               ! use the ratio of liquid to (liquid+ice) in the top layer to determine split
+               if ((h2osoi_liq(c,j)+h2osoi_ice(c,j)) > 0._r8) then
+                  qflx_liqevap_from_top_layer(p) = max(qflx_ev_snow(p)*(h2osoi_liq(c,j)/ &
+                       (h2osoi_liq(c,j)+h2osoi_ice(c,j))), 0._r8)
+               else
+                  qflx_liqevap_from_top_layer(p) = 0._r8
+               end if
+               qflx_solidevap_from_top_layer(p) = qflx_ev_snow(p) - qflx_liqevap_from_top_layer(p)
+            else
+               if (t_grnd(c) < tfrz) then
+                  qflx_soliddew_to_top_layer(p) = abs(qflx_ev_snow(p))
+               else
+                  qflx_liqdew_to_top_layer(p) = abs(qflx_ev_snow(p))
+               end if
+            end if
+
+         else ! Urban columns
+
+            if (qflx_evap_soi(p) >= 0._r8) then
+               ! for evaporation partitioning between liquid evap and ice sublimation,
+               ! use the ratio of liquid to (liquid+ice) in the top layer to determine split
+               if ((h2osoi_liq(c,j)+h2osoi_ice(c,j)) > 0._r8) then
+                  qflx_liqevap_from_top_layer(p) = max(qflx_evap_soi(p)*(h2osoi_liq(c,j)/ &
+                       (h2osoi_liq(c,j)+h2osoi_ice(c,j))), 0._r8)
+               else
+                  qflx_liqevap_from_top_layer(p) = 0._r8
+               end if
+               qflx_solidevap_from_top_layer(p) = qflx_evap_soi(p) - qflx_liqevap_from_top_layer(p)
+            else
+               if (t_grnd(c) < tfrz) then
+                  qflx_soliddew_to_top_layer(p) = abs(qflx_evap_soi(p))
+               else
+                  qflx_liqdew_to_top_layer(p) = abs(qflx_evap_soi(p))
+               end if
+            end if
+
+         end if
+
+      end do
```

Comment:
This block of code was moved as-is from below: it looks safe to do this partitioning earlier in the subroutine, and doing so lets us directly adjust the partitioning later.